lib/livebook/apps.ex

```diff
@@ -79,13 +80,11 @@ defmodule Livebook.Apps do
   @spec authorized?(App.t(), Livebook.Users.User.t()) :: boolean()
   def authorized?(app, user)
 
-  def authorized?(%{app_spec: %Livebook.Apps.TeamsAppSpec{}}, %{restricted_apps_groups: []}),
-    do: false
+  def authorized?(%{app_spec: %Apps.TeamsAppSpec{}}, %{groups: [], access_type: :apps}), do: false
```

Comment:
It feels we should remove this clause because `Livebook.Hubs.TeamClient.user_app_access?` will check for it anyway? :)